protocol/contracts/v2/TreasuryReservesVault.sol

```diff
@@ -80,395 +95,585 @@ contract TreasuryReservesVault is ITreasuryReservesVault, TempleElevatedAccess {
-        address _stableToken,
-        address _internalDebtToken,
         address _tpiOracle
     ) TempleElevatedAccess(_initialRescuer, _initialExecutor)
     {
-        templeToken = ITempleERC20Token(_templeToken);
-        stableToken = IERC20(_stableToken);
-        internalDebtToken = ITempleDebtToken(_internalDebtToken);
         tpiOracle = ITreasuryPriceIndexOracle(_tpiOracle);
     }
 
-    function setBaseStrategy(address _baseStrategy) external override onlyElevatedAccess {
-        baseStrategy = ITempleBaseStrategy(_baseStrategy);
-
-        // Ensure the new base strategy has it's TRV set as this contract.
-        if (address(baseStrategy.treasuryReservesVault()) != address(this)) 
-            revert CommonEventsAndErrors.InvalidAddress(address(baseStrategy.treasuryReservesVault()));
-
-        // Also check the API version matches.
-        if (keccak256(abi.encodePacked(baseStrategy.apiVersion())) != keccak256(abi.encodePacked(API_VERSION)))
-            revert CommonEventsAndErrors.InvalidAddress(address(baseStrategy.treasuryReservesVault()));
+    /**
+     * @notice Set the borrow token configuration. 
+     * @dev This can either add a new token or update an existing token.
+     */
+    function setBorrowToken(
+        IERC20 token, 
+        address baseStrategy,
+        uint256 baseStrategyWithdrawalBuffer,
+        uint256 baseStrategyDepositThreshold,
+        address dToken
+    ) external override onlyElevatedAccess {
+        if (address(token) == address(0)) revert CommonEventsAndErrors.InvalidAddress();
+        if (address(dToken) == address(0)) revert CommonEventsAndErrors.InvalidAddress();
+        if (baseStrategyDepositThreshold < baseStrategyWithdrawalBuffer) revert CommonEventsAndErrors.InvalidParam();
+
+        emit BorrowTokenSet(address(token), baseStrategy, baseStrategyWithdrawalBuffer, baseStrategyDepositThreshold, dToken);
+        borrowTokens[token] = BorrowTokenConfig({
+            baseStrategy: ITempleBaseStrategy(baseStrategy),
+            baseStrategyWithdrawalBuffer: baseStrategyWithdrawalBuffer,
+            baseStrategyDepositThreshold: baseStrategyDepositThreshold,
+            dToken: ITempleDebtToken(dToken)
+        });
+        _borrowTokenSet.add(address(token));
+    }
 
-        emit BaseStrategySet(_baseStrategy);
+    /**
+     * @notice Remove the borrow token configuration. 
+     */
+    function removeBorrowToken(
+        IERC20 token
+    ) external override onlyElevatedAccess {
+        if (!_borrowTokenSet.contains(address(token))) revert BorrowTokenNotEnabled();
+        emit BorrowTokenRemoved(address(token));
+        delete borrowTokens[token];
+        _borrowTokenSet.remove(address(token));
     }
 
     /**
      * @notice Set the Treasury Price Index (TPI) Oracle
      */
     function setTpiOracle(address newTpiOracle) external override onlyElevatedAccess {
+        if (address(newTpiOracle) == address(0)) revert CommonEventsAndErrors.InvalidAddress();
```

Comment:
nit: after setting tpi oracle, check `tpiData.currentTpi > 0`

Reply:
Fixed in https://github.com/TempleDAO/temple/pull/814/commits/6745674e1060077e4e8093751dd1b3149267d47f